web-marketplace/src/pages/CreateOrganization/steps/OrganizationProfileStep.tsx

```diff
@@ -0,0 +1,208 @@
+      if (initialValues?.name && initialValues.name.trim().length > 0) {
+        setTransferHandled(true);
+        setShowTransferModal(false);
+      }
+    }, [initialValues?.name]);
+
+    const handleCloseTransferModal = useCallback(() => {
+      setTransferHandled(true);
+      setShowTransferModal(false);
+    }, []);
+
+    const handleTransferProfile = useCallback(() => {
+      if (!activeAccount) {
+        handleCloseTransferModal();
+        return;
+      }
+
+      const fallbackName = activeAccount.name?.trim() || _(DEFAULT_NAME);
+      const nextValues: Partial<EditProfileFormSchemaType> = {
+        name: fallbackName,
+        description: activeAccount.description?.trim() ?? '',
+        profileImage: activeAccount.image || DEFAULT_PROFILE_COMPANY_AVATAR,
+        backgroundImage: activeAccount.bgImage || DEFAULT_PROFILE_BG,
+        websiteLink: activeAccount.websiteLink?.trim() ?? '',
+        twitterLink: activeAccount.twitterLink?.trim() ?? '',
+        profileType:
+          activeAccount.type === AccountType.Organization
+            ? AccountType.Organization
+            : undefined,
+      };
+
+      onTransferProfile({ nextValues });
+      setTransferHandled(true);
+      setShowTransferModal(false);
+    }, [activeAccount, handleCloseTransferModal, onTransferProfile, _]);
+
+    const handleSubmit = useCallback(
+      async (values: EditProfileFormSchemaType) => {
+        if (hasUnfinishedOrganization && daoAddress) {
+          setTransferHandled(true);
+          setShowTransferModal(false);
+          const payload: OrganizationMultiStepData = {
+            ...values,
+            dao: data?.dao ?? {
+              daoAddress,
+              organizationId,
+            },
+          };
+          handleSaveNext(payload);
+          return;
+        }
+
+        try {
+          if (!activeAccountId) {
+            throw new Error(_(CREATE_ORG_ACTIVE_ACCOUNT_REQUIRED_ERROR));
+          }
+
+          const organizationIdValue = organizationId ?? uuidv4();
+          setOrganizationId(organizationIdValue);
+
+          const daoResult = await createDao({
+            name: values.name,
+            description: values.description,
+            profileImage: values.profileImage,
+            backgroundImage: values.backgroundImage,
+            websiteLink: values.websiteLink,
+            twitterLink: values.twitterLink,
+            organizationId: organizationIdValue,
+            currentAccountId: activeAccountId,
+          });
+
+          setDaoAddress(daoResult.daoAddress);
+          setOrganizationId(daoResult.organizationId);
+          setTransferHandled(true);
+          setShowTransferModal(false);
+
+          const payload: OrganizationMultiStepData = {
+            ...values,
+            dao: daoResult,
+          };
+          handleSaveNext(payload);
+        } catch (error) {
+          if (error instanceof Error) {
+            throw error;
+          }
+          throw new Error(
```

Comment:
let's use setErrorBanner too here
please review all the places you throw new Error and replace